apps/canvas/canvas.star

```diff
@@ -107,21 +109,24 @@ def get_remote_assignments(api_token, course_id):
            (time.parse_time(assignment["due_at"]) - time.now()).hours < 200
     ]
 
-def cache_assignments(course_id, assignments):
+def cache_assignments(course_id, assignments, api_token):
     cache_string = ";".join([a[0] + "," + a[1] for a in assignments])
-    cache.set(str(course_id), cache_string, ttl_seconds = 300)
+    cache.set(str(course_id) + "-" + api_token, cache_string, ttl_seconds = 300)
 
 def get_events(api_token, course_id):
-    assignment_data = get_cached_assignments(course_id)
+    assignment_data = get_cached_assignments(course_id, api_token)
     if assignment_data == None:
         print("Not Using Cached Classes for" + str(course_id))
         assignment_data = get_remote_assignments(api_token, course_id)
-        cache_assignments(course_id, assignment_data)
+        cache_assignments(course_id, assignment_data, api_token)
     return assignment_data, 0
 
 def main(config):
     api_token = config.get("msg", "")
 
+    if api_token == "":
```

Comment:
In this case, it would be great if the app could produce a default graphic showing what it'll look like once it's set up. This would allow users to see a preview of the app before they decided if they want to go hunting for their API key.